bundle/src/main/java/com/adobe/acs/commons/mcp/impl/processes/BulkPageTagger.java

```diff
@@ -0,0 +1,268 @@
+        // Nothing to be done here.
+
+    }
+
+
+    @Override
+    public void buildProcess(ProcessInstance instance, ResourceResolver rr) throws LoginException {
+        report.setName(instance.getName());
+        instance.getInfo().setDescription("Bulk Tag AEM content Pages");
+        instance.defineCriticalAction("Parse Excel File", rr, this::parseExcel);
+        instance.defineCriticalAction("Add Tags to Content Pages", rr, this::tagPages);
+
+    }
+
+
+    /**
+     * The Page tag mapping dataStructure.
+     */
+    transient volatile HashMap<String, String> pageTagMapping = new LinkedHashMap<>();
+
+    /**
+     * Parse input excel.
+     *
+     * @param manager the manager
+     * @throws Exception the exception
+     */
+    @SuppressWarnings({"squid:S3776", "squid:S1141"})
+    public void parseExcel(ActionManager manager) throws Exception {
+        manager.withResolver(rr -> {
+            final XSSFWorkbook workbook = new XSSFWorkbook(excelFile);
+            final XSSFSheet sheet = workbook.getSheetAt(0);
+
+            final Iterator<Row> rows = sheet.rowIterator();
+            final String tagsRootPath = new TagCreator.TagRootResolver(rr).getTagsLocationPath();
+
+
+            if (tagsRootPath == null) {
+                recordAction(ReportRowSatus.FAILED_TO_PARSE,
+                        "Abandoning Tag parsing. Unable to determine AEM Tags root (/content/cq:tags vs /etc/tags). Please ensure the path exists and is accessible by the user running Tag Creator.", "N/A");
+                return;
+            }
+
+            while (rows.hasNext()) {
+                final Row row = rows.next();
+                if (row.getCell(0) == null) {
+                    break;
+                }
+                if (row.getRowNum() != 0 && row.getCell(0) != null) {
+                    pageTagMapping.put(row.getCell(0).getStringCellValue(), row.getCell(1).getStringCellValue());
+                }
+            }
+
+        });
+
+    }
+
+
+    /**
+     * Tag pages from the excel file with cq:tags.
+     *
+     * @param manager the manager
+     * @throws Exception the exception
+     */
+    @SuppressWarnings({"squid:S3776", "squid:S1141"})
+    public void tagPages(ActionManager manager) throws Exception {
+
+        manager.withResolver(rr -> {
+
+            pageTagMapping.forEach((key, value) -> {
+                BulkPageTagger.ReportRowSatus status;
+                Resource resource = rr.getResource(key);
+                if (resource != null) {
+                    Page page = resource.adaptTo(Page.class);
+                    if (page != null) {
+                        Tag[] existingPageTags = page.getTags();
+                        String[] tagIds = Stream.of(existingPageTags)
+                                .map(Tag::getTagID)
+                                .toArray(String[]::new);
+                        Set<String> updatedTags = new HashSet<>(Arrays.asList(value.split(";")));
```

Comment:
Can you support `\n` as a tag separator along with semicolon? There are other MCP jobs  that read tags from a spreadsheet, like CF Importer, and they assume tags come from a multi-line cell. See attached sample. 
[ACS.Commons.Bulk.Page.tagger.xlsx](https://github.com/Adobe-Consulting-Services/acs-aem-commons/files/12465076/ACS.Commons.Bulk.Page.tagger.xlsx)

Reply:
@YegorKozlov , Should we add this tag separator as an option that the user can choose from the MCP console or handling this in the code is sufficient ?

Reply:
Let's support both. String.split accepts a regex and `split(";|\\n")` should work.

Reply:
@YegorKozlov Done! please review. Thanks.